XLS-0070d-credentials/README.md

```diff
@@ -0,0 +1,573 @@
+
+### 2.2. Account Deletion
+
+The `Credential` object is not a [deletion blocker](https://xrpl.org/docs/concepts/accounts/deleting-accounts/#requirements).
+
+## 3. On-Ledger Object: `DepositPreauth`
+
+The `DepositPreauth` object tracks a preauthorization from one account to another. This object already exists on the XRPL, but is being extended as a part of this spec to also support credential preauthorization.
+
+### 3.1. Fields
+<details>
+<summary>
+
+As a reference, [here](https://xrpl.org/docs/references/protocol/ledger-data/ledger-entry-types/depositpreauth/#depositpreauth-fields) are the existing fields for the `DepositPreauth` object.
+</summary>
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`Account`|✔️|`string`|`AccountID`|The account that granted the preauthorization (the destination of the preauthorized payments).|
+|`Authorize`|✔️|`string`|`AccountID`|The account that received the preauthorization (the sender of the preauthorized payments).|
+|`LedgerEntryType`|✔️|`string`|`UInt16`|The value `0x0070`, mapped to the string `"DepositPreauth"`, indicates that this is a `DepositPreauth` object.|
+|`OwnerNode`|✔️|`string`|`UInt64`|A hint indicating which page of the sender's owner directory links to this object, in case the directory consists of multiple pages. Note: The object does not contain a direct link to the owner directory containing it, since that value can be derived from the `Account.PreviousTxnID`.|
+|`PreviousTxnID`|✔️|`string`|`Hash256`|The identifying hash of the transaction that most recently modified this object.|
+|`PreviousTxnLgrSeqNumber`|✔️|`number`|`UInt32`|The index of the ledger that contains the transaction that most recently modified this object.|
+</details>
+
+We propose these modifications:
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`Authorize`| |`string`|`AccountID`|The account that received the preauthorization. (The sender of the preauthorized payments.)|
+|`AuthorizeCredentials`| |`array`|`STArray`|The credential that received the preauthorization. (Any account with this credential can send preauthorized payments).|
+
+#### 3.1.1. Object ID
+
+The ID of this object will be either a hash of the `Account` and `Authorize` fields (as it currently is), or a hash of the `Account` and the contents of `AuthorizeCredentials` fields, combined with the unique space key for `DepositAuth` objects: `0x0070`.
+
+#### 3.1.2. `Authorize`
+
+This field is now optional, and a valid `DepositPreauth` object must either have the `Authorize` field or the `AuthorizeCredentials` field.
+
+#### 3.1.3. `AuthorizeCredentials`
+
+This field is an array of inner objects. The contents of these inner objects determine the credential(s) that are accepted.
```

Comment:
> Does the ordering of the items in the array matter?

They have to be sorted in the object. Otherwise, you'd run into issues where you provided the right group of credentials but in the wrong order, so they weren't accepted.

> Also, just to confirm, if you do have multiple DepositPreauth entries that could allow an action, it doesn't really matter which one actually did authorize it... right?

Correct. Though you'd have only two possible objects that could do so - the one with the credentials (if `CredentialIDs` is included as a field) and the one with the account.